### PR TITLE
[PM-8027] Inline menu appears within input fields that do not relate to user login

### PR DESCRIPTION
**Pull request by cagonzalezcs**
## 🎟️ Tracking

https://bitwarden.atlassian.net/browse/PM-8027

## 📔 Objective

The goal of this ticket is to refine the manner in which we position the inline menu within a web page. The initial release of the inline menu had a very rudimentary approach to validating whether we should place the menu on an input field.

This newer approach takes into consideration the `autocomplete` attribute of the form field element, as well as other design heuristics that surround the form field.  

As a result of these changes, we consider a username field to be part of a login form as:
- A field that contains keywords that indicate that the field is a username field
- A field that is structured with no more than one other password field within the same form or the same page if the filed is formless
- A field that is not surrounded by other fields if a matching password field is not found
- A field that does not contain a disabled `autocomplete` value of `off` or `false` if a matching password field is not found
- A field that has a matching password field, even if the field contains a disabled `autocomplete` value.

Similarly, we consider a password field to be part of a login form if:
- The password field has an `autocomplete` value of `current-password`. We disqualify all fields that contain a `new-password` `autocomplete` value.
- If the password field is matched by a single username field on the same page
- If the password field has no form parent, but is the only password field on the page and it does not have a disabled `autcomplete` value

These rules likely will change as we refine the process of qualifying login form fields, however we'd like to use this implementation as a starting point.

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Protected functional changes with optionality (feature flags)
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation (Confluence, contributing docs) or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

### CODE REVIEW

**Comment by cagonzalezcs** [comment]
BIT - https://github.com/bitwarden/browser-interactions-testing/actions/runs/9369331226

File: apps/browser/src/autofill/services/inline-menu-field-qualification.service.ts [hunk 1]
```diff
@@ -0,0 +1,395 @@
...
+  /**
+   * Validates the provided field as a new password field.
+   *
+   * @param field - The field to validate
+   */
+  private isNewPasswordField = (field: AutofillField): boolean => {
+    if (field.autoCompleteType === "current-password") {
+      return false;
+    }
+
```
Comment (jprusik) [inline apps/browser/src/autofill/services/inline-menu-field-qualification.service.ts:227]: We could also shortcut this if the autocomplete type is explicitly a new password:
```suggestion

    if (field.autoCompleteType === "new-password") {
      return true;
    }

```

**Review by jprusik — APPROVED** [review]

File: apps/browser/src/autofill/services/inline-menu-field-qualification.service.ts (hunk 1 above)
Comment (jprusik) [reply apps/browser/src/autofill/services/inline-menu-field-qualification.service.ts:241]: > Hmm... so this pulls us closer to "trusting" the intent of a developer's markup... would that be desired?

I see the values of "new-password" and "current-password" more as descriptors than prescribing behaviour, whereas "off" is not helpful as a descriptor and arguably not trustworthy as a prescriber of behaviour to carry out user-intent.

**Review by jprusik — DISMISSED** [review]

**Review by jprusik — APPROVED** [review]
LGTM &  BIT run:
https://github.com/bitwarden/browser-interactions-testing/actions/runs/9550184650

**Pull request merged**